zipline/assets/assets.py

```diff
@@ -678,3 +728,113 @@ class AssetConvertible(with_metaclass(ABCMeta)):
 
 class NotAssetConvertible(ValueError):
     pass
+
+
+class AssetFinderCachedEquities(AssetFinder):
+    """
+    An extension to AssetFinder that loads all equities from equities table
+    into memory and overrides the methods that lookup_symbol uses to look up
+    those equities.
+    """
+    def __init__(self, engine):
+        super(AssetFinderCachedEquities, self).__init__(engine)
+        self.fuzzy_symbol_hashed_equities = {}
+        self.company_share_class_hashed_equities = {}
+        self.hashed_equities = sa.select(self.equities.c).execute().fetchall()
+        self.load_hashed_equities()
+
+    def load_hashed_equities(self):
+        """
+        Populates two maps - fuzzy symbol to list of equities having that
+        fuzzy symbol and company symbol/share class symbol to list of
+        equities having that combination of company symbol/share class symbol.
+        """
+        for equity in self.hashed_equities:
+            company_symbol = equity['company_symbol']
+            share_class_symbol = equity['share_class_symbol']
+            fuzzy_symbol = equity['fuzzy_symbol']
+            asset = self.convert_row_to_equity(equity)
+            self.company_share_class_hashed_equities.setdefault(
+                (company_symbol, share_class_symbol),
+                []
+            ).append(asset)
+            self.fuzzy_symbol_hashed_equities.setdefault(
+                fuzzy_symbol, []
+            ).append(asset)
+
+    def convert_row_to_equity(self, equity):
+        """
+        Converts a SQLAlchemy equity row to an Equity object.
+        """
+        data = dict(equity.items())
+        _convert_asset_timestamp_fields(data)
+        asset = Equity(**data)
+        return asset
+
+    def get_fuzzy_candidates(self, fuzzy_symbol):
```

Comment:
Since these methods are internal API, you can prepend them with _ as in 
```
def _get_fuzzy_candidates(self, fuzzy_symbol):
```
so that it is clear, when working with an AssetFinder, that these methods are not meant to be accessed directly.